extension/scripts/main.js

```diff
@@ -131,7 +150,14 @@ chrome.runtime.onMessage.addListener(function (request) {
 
 	if (request.method === 'startTests') {
 		if (!widget) loadWidget();
-		beginTests();
+		beginTests()
+			.then(testResults => {
+				testResults.forEach(result => {
```

Comment:
Can you update the results after each test.

Some tests may never resolve unless they fail. E.g. A user could add a scroll listener at any point in the life time of the page.